File: PHP/8php 2/is_divisible.php
<?php
/*
|--------------------------------------------------------------------------
| Is n divisible by x and y?
|--------------------------------------------------------------------------
|
*/

function is_divisible($n, $x, $y) {
   return $n % $x == 0 && $n % $y == 0;
}


function is_divisible($n, $x, $y) {
   return $n % $x == 0 and $n % $y == 0 ? true : false;
}


function is_divisible(int $n, int $x, int $y) : bool {
   return !($n % $x || $n % $y);
}


print_r(is_divisible(3, 3, 4). ' ' . false . '
');
print_r(is_divisible(12, 3, 4). ' ' . true . '
');
print_r(is_divisible(8, 3, 4). ' ' . false . '
');
print_r(is_divisible(48, 3, 4). ' ' . true . '
');

File: Python/8py 2/is_divisible.py
"""
|--------------------------------------------------------------------------
| Is n divisible by x and y?
|--------------------------------------------------------------------------
|
"""

def is_divisible(n,x,y):
   return (n%x)==0 and (n%y)==0

   
def is_divisible(n,x,y):
   return not (n%x or n%y)

   
def is_divisible(n,x,y):
   return n % x + n % y == 0 

   
def is_divisible(n, x, y):
   return not n % x and not n % y

   
from math import gcd; is_divisible = lambda n, x, y: not n%(x*y//(gcd(x, y)))


print(is_divisible(3,2,2),False)
print(is_divisible(3,3,4),False)
print(is_divisible(12,3,4),True)
print(is_divisible(8,3,4),False)